piptools/utils.py

```diff
@@ -48,13 +48,33 @@ def make_install_requirement(name, version, extras, constraint=False):
     )
 
 
+def is_url_requirement(ireq, repository=None):
+    """
+    Finds if a requirement is a URL
+    """
+    if (
+        not ireq.link
+        or "files.pythonhosted.org" in ireq.link.url
```

Comment:
It could be valid for the link like this `https://example.com/storage/files.pythonhosted.org.zip`. Maybe it's better to check the `netloc`?